src/os/bluestore/BlueStore.cc

```diff
@@ -16793,21 +16801,23 @@ int BlueStore::_do_alloc_write(
   if (wctx->compress) {
     c = select_option(
```

Comment:
I wonder if all this logic (and in similar `select_option` cases should not be handled in `BlueStore::set_collection_opts`.
All errors and complaints will be handled and printed there.
What will remain will be just
`int compression_alg;`,
and will be used directly.